include/sliding_window/split/reference_segments.hpp

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/io/sequence_file/all.hpp>
+
+#include <iostream>
+#include <fstream>
+
+namespace sliding_window
+{
+
+class reference_segments {
+    public:
+        class segment {
+            public:
+                size_t bin;
+                std::string ref_id;
+                size_t start;
+                size_t len;
+
+            segment(size_t b, std::string id, size_t s, size_t l)
+            {
+                bin = b;
+                ref_id = id;
+                start = s;
+                len = l;
+            }
+        };
+
+        std::vector<segment> members;
+        size_t default_len;
+
+        void add_segment(size_t b, std::string id, size_t s, size_t l)
+        {
+            segment seg(b, id, s, l);
+            members.push_back(seg);
+        }
+
+        reference_segments(size_t segment_len, size_t overlap, auto & reference_sequences)
+        {
+            default_len = segment_len;
+            size_t i = 0;
+            for (const auto & seq : reference_sequences)
+            {
+                size_t start = 0;
+
+                // only one segment per reference sequence
+                if (start + segment_len >= seq.len)
+                {
+                    add_segment(i, seq.id, start, seq.len);
+                    i++;
+                }
+
+                // many segments per reference sequence
+                else
+                {
+                    add_segment(i, seq.id, start, segment_len);
+                    i++;
+                    while (start + segment_len < seq.len)
+                    {
+                        start = start + segment_len - overlap;
+                        add_segment(i, seq.id, start, segment_len);
+                        i++;
+                    }
+                    add_segment(i, seq.id, start, seq.len - start);
+                    i++;
+                }
+            }
```

Comment:
Normally you don't do computation in a constructor.

Reply:
I would at least also put this into an extra function, to declutter the constructor